x-pack/plugins/cloud_security_posture/public/components/csp_page_template.tsx

```diff
@@ -0,0 +1,175 @@
+ * 2.0.
+ */
+import React from 'react';
+import type { UseQueryResult } from 'react-query';
+import { NavLink } from 'react-router-dom';
+import { EuiEmptyPrompt, EuiErrorBoundary, EuiTitle } from '@elastic/eui';
+import { FormattedMessage } from '@kbn/i18n-react';
+import {
+  KibanaPageTemplate,
+  type KibanaPageTemplateProps,
+} from '../../../../../src/plugins/kibana_react/public';
+import { allNavigationItems } from '../common/navigation/constants';
+import type { CspNavigationItem } from '../common/navigation/types';
+import { CLOUD_SECURITY_POSTURE } from '../common/translations';
+import { CspLoadingState } from './csp_loading_state';
+import { LOADING, PACKAGE_NOT_INSTALLED_TEXT, DEFAULT_NO_DATA_TEXT } from './translations';
+import { useCisKubernetesIntegration } from '../common/api/use_cis_kubernetes_integration';
+import { useCISIntegrationLink } from '../common/navigation/use_navigate_to_cis_integration';
+
+interface QueryError {
+  body: {
+    error: string;
+    message: string;
+    statusCode: number;
+  };
+  name: string;
+}
+
+const activeItemStyle = { fontWeight: 700 };
+
+export const getSideNavItems = (
+  navigationItems: Record<string, CspNavigationItem>
+): NonNullable<KibanaPageTemplateProps['solutionNav']>['items'] =>
+  Object.entries(navigationItems)
+    .filter(([_, navigationItem]) => !navigationItem.disabled)
+    .map(([id, navigationItem]) => ({
+      id,
+      name: navigationItem.name,
+      renderItem: () => (
+        <NavLink to={navigationItem.path} activeStyle={activeItemStyle}>
+          {navigationItem.name}
+        </NavLink>
+      ),
+    }));
+
+const DEFAULT_PAGE_PROPS: KibanaPageTemplateProps = {
+  solutionNav: {
+    name: CLOUD_SECURITY_POSTURE,
+    items: getSideNavItems({
+      dashboard: allNavigationItems.dashboard,
+      findings: allNavigationItems.findings,
+      benchmark: allNavigationItems.benchmarks,
+    }),
+  },
+  restrictWidth: false,
+};
+
+export const DEFAULT_NO_DATA_CONFIG: KibanaPageTemplateProps['noDataConfig'] = {
+  pageTitle: DEFAULT_NO_DATA_TEXT.PAGE_TITLE,
+  solution: DEFAULT_NO_DATA_TEXT.SOLUTION,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {},
+};
+
+const getPackageNotInstalledConfig = (
+  cisIntegrationLink: string
+): KibanaPageTemplateProps['noDataConfig'] => ({
+  pageTitle: PACKAGE_NOT_INSTALLED_TEXT.PAGE_TITLE,
+  solution: PACKAGE_NOT_INSTALLED_TEXT.SOLUTION,
+  // TODO: Add real docs link once we have it
+  docsLink: 'https://www.elastic.co/guide/index.html',
+  logo: 'logoSecurity',
+  actions: {
+    elasticAgent: {
+      href: cisIntegrationLink,
+      title: PACKAGE_NOT_INSTALLED_TEXT.BUTTON_TITLE,
+      description: PACKAGE_NOT_INSTALLED_TEXT.DESCRIPTION,
+    },
+  },
+});
+
+const DefaultLoading = () => <CspLoadingState>{LOADING}</CspLoadingState>;
+
+const DefaultError = (error: QueryError) => (
```

Comment:
for now those are not `React.FC<{error: TError}>` but a `(error: TError) => React.ReactNode`
I find most `render` props to works in this same way allowing you to get use them like:
`render: (data) => <div>{data}</>`
instead of 
`render: ({data}) => <div>{data}</>`
small difference but I prefer the first one since I see it more often when using those myself. please say so if you think this change is important.